### ISSUE TITLE
The binary version of its metadata is 1.9.0, expected version is 1.x.x

### ISSUE THREAD

**Issue by emirua** [auth0/auth0-flutter#310]
### Checklist

- [X] The issue can be reproduced in the [auth0_flutter sample app](https://github.com/auth0-samples/auth0-flutter-samples/tree/main/sample) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/auth0-flutter/tree/main/auth0_flutter#readme), [Examples](https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://pub.dev/documentation/auth0_flutter/latest/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-flutter/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

This is a shared problem with the auth0 android repo and the last update https://github.com/auth0/Auth0.Android/releases/tag/2.10.2 

Flutter is actually not currently supporting kotlin v1.9.0 so this is creating an issue which is not evident at first since we are not updating any dependency in pub but the way this library is managing the dependency using 
`implementation 'com.auth0.android:auth0:2a.+'` is updating to the latest version in Android and making this problem hard to identify.

Please stop using the '+' to manage the android dependency.
 

### Reproduction

1. implement library latest version.
2. try running the project.
3. the project will fail.

### Additional context

'com.auth0.android:auth0:2.10.1' works well, but 2.10.2 breaks the project.

If I manually change this in the library gradle, the project runs, but this fix is not possible in CI/CD workflows, so I think this fix is really important

### auth0_flutter version

1.2.1

### Flutter version

3.13.6

### Platform

Android

### Platform version(s)

2.10.2

**Comment by poovamraj** [comment]
@emirua I agree that using `+` is not great for the SDK. This has been raised internally.

But the new release was to handle CVE-2023-3635 in okio. This means it is better to use the latest version. Flutter prompts to fix the issues and you can see the reference here on how to fix the issue you are facing.

```

┌─ Flutter Fix ────────────────────────────────────────────────────────────────┐
│ [!] Your project requires a newer version of the Kotlin Gradle plugin.       │
│ Find the latest version on                                                   │
│ https://kotlinlang.org/docs/releases.html#release-details, then update       │
│ /Users/poovamrajthanganadarthiagarajan/repositories/auth0-flutter/auth0_flut │
│ ter/example/android/build.gradle:                                            │
│ ext.kotlin_version = '<latest-version>'                                      │
└──────────────────────────────────────────────────────────────────────────────┘
```

```
 Kotlin: Unsupported version of com.android.tools.build:gradle plugin: version 4.2.2 or higher should be used with kotlin-android plugin
 ```
 
 Or you can see the change in this [commit](https://github.com/auth0/auth0-flutter/commit/193c7260a5084f73e4d2aa11a60afd3f19663c4b) and it should fix the build issues you would face.

**Comment by emirua** [comment]
I see this working in a regular flutter application, however this fix doesn't work in 'add to app' implementations since the kotlin version there is set to 1.7.10 as of today in the stable channel. It is possible to change the local  configuration to 1.9.10 but this fix is not possible in CI/CD workflows :(.

**Comment by Widcket** [comment]
The fix is now out in [v1.3.0](https://github.com/auth0/auth0-flutter/releases/tag/af-v1.3.0).
